feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.spec.ts

```diff
@@ -0,0 +1,242 @@
+    fixture = TestBed.createComponent(AddedToCartToastComponent);
+    component = fixture.componentInstance;
+    component.timeout = 3000;
+    component.scrollEventUnlistener = () => {};
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call the active cart service', () => {
+    const spy = spyOn(activeCartService, 'getLastEntry').and.callThrough();
+    component.ngOnInit();
+    component.addToast(mockEvent);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should set the class on init', () => {
+    component.ngOnInit();
+    expect(component.baseClass).toBe('cx-added-to-cart-toast');
+  });
+
+  it('should have the product name', () => {
+    component.ngOnInit();
+    const el = fixture.debugElement.query(By.css('h3'));
+    expect(el.nativeElement.innerHTML).toContain(mockCartToastItem.productName);
+  });
+
+  it('should have the quantity added', () => {
+    component.ngOnInit();
+    const el = fixture.debugElement.query(
+      By.css('div.added-to-cart-toast-title')
+    );
+    expect(el.nativeElement.innerHTML).toContain(mockCartToastItem.quantity);
+  });
+
+  it('should have the unit price', () => {
+    component.ngOnInit();
+    const el = fixture.debugElement.query(By.css('span.price'));
+    expect(el.nativeElement.innerHTML).toContain(mockCartToastItem.unitPrice);
+  });
+
+  describe('isHeaderFixed', () => {
+    it('should be fixed', () => {
+      component.headerElement = document.createElement('header');
+      component.headerElement.style.position = 'fixed';
+      const res = component.isHeaderFixed();
+      expect(res).toBeTruthy();
+    });
+
+    it('should not be fixed', () => {
+      component.headerElement = document.createElement('header');
+      component.headerElement.style.position = 'absolute';
+      const res = component.isHeaderFixed();
+      expect(res).not.toBeTruthy();
+    });
+  });
+
+  describe('isHeaderInView', () => {
+    it('should be in view', () => {
+      component.headerElement = document.createElement('header');
+      spyOn(component.headerElement, 'getBoundingClientRect').and.returnValue(
+        new DOMRect(100, 2000, 100, 100)
+      );
+      const res = component.isHeaderInView();
+      expect(res).toBeTruthy();
+    });
+    it('should not be in view', () => {
+      component.headerElement = document.createElement('header');
+      spyOn(component.headerElement, 'getBoundingClientRect').and.returnValue(
+        new DOMRect(100, -2000, 100, 100)
+      );
+      const res = component.isHeaderInView();
+      expect(res).not.toBeTruthy();
+    });
+  });
+
+  describe('Toast Positions', () => {
+    describe('Toast Opening', () => {
+      it('should be fixed when the header is fixed', () => {
+        spyOn(component, 'isHeaderFixed').and.returnValue(true);
+        let positionClasses = component.getToastStyles(
```

Comment:
we could declare `positionClasses` only once in the describe block and only do the assignments in test cases